shell/platform/darwin/ios/ios_platform_surface_gl.mm

```diff
@@ -0,0 +1,61 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/darwin/ios/ios_platform_surface_gl.h"
+#include "flutter/common/threads.h"
+#include "flutter/lib/ui/painting/resource_context.h"
+#include "flutter/shell/platform/darwin/ios/framework/Source/vsync_waiter_ios.h"
+#include "third_party/skia/include/core/SkSurface.h"
+#include "third_party/skia/include/gpu/GrBackendSurface.h"
+#include "third_party/skia/include/gpu/GrTexture.h"
+#include "third_party/skia/include/gpu/GrTypes.h"
+
+namespace shell {
+
+IOSPlatformSurfaceGL::~IOSPlatformSurfaceGL() {
+  // TODO(sigurdm, mravn): dispose cache
+}
+
+IOSPlatformSurfaceGL::IOSPlatformSurfaceGL(NSObject<FlutterPlatformSurface>* surface)
+    : surface_(surface) {
+  FXL_DCHECK(surface_);
+}
+
+sk_sp<SkImage> IOSPlatformSurfaceGL::MakeSkImage(int width, int height, GrContext* grContext) {
+  ASSERT_IS_GPU_THREAD;
+  if (cache_ == nullptr) {
+    CVReturn err = CVOpenGLESTextureCacheCreate(kCFAllocatorDefault, NULL,
+                                                [EAGLContext currentContext], NULL, &cache_);
+    if (err != noErr) {
+      FXL_LOG(WARNING) << "Failed to create GLES texture cache: " << err;
+    }
+  }
+  CVPixelBufferRef buffer = [surface_ getPixelBuffer];
```

Comment:
`getPixelBuffer` reads as `transfer-none`. So you'd need a `CFRetain` here. Also, please use a `fml::CFRef` to hold a buffer reference.

Reply:
Update: `getPixelBuffer` -> `copyPixelBuffer` and no retain.